src/clp_ffi_js/ir/StreamReader.hpp

```diff
@@ -52,6 +51,9 @@ class StreamReader {
     auto operator=(StreamReader&&) -> StreamReader& = delete;
 
     // Methods
+    [[nodiscard]] virtual auto get_ir_protocol_error_code(
```

Comment:
Why do we need to expose this method? If it's just so that the caller can determine what type of IR stream we're reading, then I think it's clearer to return some kind of enum of `Structured` and `Unstructured` (and call the method something like `get_ir_stream_type`).

Reply:
Gotcha. I'd say an enum indicating that is still clearer than returning the protocol error code.